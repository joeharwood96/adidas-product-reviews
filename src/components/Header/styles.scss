.header {
    width: vw;
    padding: 2rem;
    display: flex;
    justify-content: space-between;

    &-logo {
        width: 4rem
    }

    &-back {
        height: 2rem;
        border: none;
        border-bottom: 1px solid black;
        white-space: unset;
        text-transform: uppercase;
        background-color: white;
        display: flex;
        align-items: center;

        &-text {
            margin-left: 0.5rem;
        }

        &:hover {
            cursor: pointer;
            background-color: black;
            color: white;
        }
    }
}