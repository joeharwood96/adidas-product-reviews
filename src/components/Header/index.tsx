import React from 'react'
import './styles.scss'
import { useNavigate } from 'react-router-dom';
import { HiArrowNarrowLeft } from 'react-icons/hi'

import logo from '../../assets/Adidas_Logo.svg'
import Search from '../Search'

export default function Header() {
    const navigate = useNavigate()
    const pathname = window.location.pathname
    const page = pathname.split('/') 

    const handleClick = () => {
        navigate('/')
    }

    return (
        <div className="header">
            <img src={logo} alt="Adidas logo" className="header-logo" />
            {
                page[1] === 'review' ? 
                <button className="header-back" onClick={handleClick}>
                    <HiArrowNarrowLeft />
                    <span className="header-back-text">back</span>
                </button>
                :
                <Search />
            }
        </div>
    )
}