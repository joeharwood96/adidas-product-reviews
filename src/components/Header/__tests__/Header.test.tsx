import React from "react";
import Header from "../index";

describe("<Header />", () => {
    test("should display the header", () => {
        const component = <Header />
        expect(component).toMatchSnapshot()
    });
});