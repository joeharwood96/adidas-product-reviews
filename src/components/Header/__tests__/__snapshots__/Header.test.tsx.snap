// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should display the header 1`] = `<Header />`;
